From e750b6b4976169d95e256208421dd24c5b8e766d Mon Sep 17 00:00:00 2001
From: Richard Dai <richarddai@163.com>
Date: Tue, 9 May 2023 16:41:04 +0800
Subject: [PATCH] add openwrt riscv target

---
 Configurations/25-openwrt.conf | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/Configurations/25-openwrt.conf b/Configurations/25-openwrt.conf
index 8407bf8..4286876 100644
--- a/Configurations/25-openwrt.conf
+++ b/Configurations/25-openwrt.conf
@@ -44,6 +44,9 @@ my %targets = (
     "linux-x86_64-openwrt" => {
         inherit_from    => [ "linux-x86_64", "openwrt" ],
     },
+    "linux-riscv64-openwrt" => {
+        inherit_from    => [ "linux64-riscv64", "openwrt" ],
+    },
 
 ### Basic default option
     "linux-generic32-openwrt" => {
-- 
2.34.1

